File: 0901-1000/0977-Squares of a Sorted Array/0977-Squares of a Sorted Array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& A) {
        vector<int> result(A.size());
        int left = 0, right = A.size() - 1;
        for (int i = A.size() - 1; i >= 0; --i) {
            if (A[left] * A[left] <= A[right] * A[right]) {
                result[i] = A[right] * A[right];
                --right;
            }
            else {
                result[i] = A[left] * A[left];
                ++left;
            }
        }

        return result;
    }
};

File: 0901-1000/0977-Squares of a Sorted Array/0977-Squares of a Sorted Array.py
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        result = [0] * len(A)
        left = 0
        right = len(A) - 1
        for i in range(len(A) - 1, -1, -1):
            if A[left] ** 2 <= A[right] ** 2:
                result[i] = A[right] ** 2
                right -= 1
            else:
                result[i] = A[left] ** 2
                left += 1

        return result
